automated-comments/go/two-fer/strings_join_used_for_concatenation.md

```diff
@@ -1,2 +1,3 @@
-Don't use `strings.Join` if the data is not of type `[]string`.
-Use `fmt.Sprintf` instead. It's a very powerful way of building up strings from multiple pieces.
+`strings.Join` is only normally used if the data is of type `[]string`.
+In other situations, it is normal to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.
+Could you try updating your solution to use that?
```

Comment:
Maybe be explicit about what to use here:

```suggestion
Could you try updating your solution to use `fmt.Sprintf`?
```

@kotp In other comments, you've suggested to remove the "you" part, which I like. Do you have any suggestion how to apply it here?

Reply:
I feel this is a different 'you', and totally okay to use here.

Reply:
I often phrase this as "Would you like to try...?" That makes it clear that this isn't a _correction_, as such; it's an alternative suggestion which may be worth exploring.

(Of course, the answer may be "No, I wouldn't". That's never happened yet, though.)

Reply:
> I feel this is a different 'you', and totally okay to use here.

Yes! Agreed.

> Would you like to try..

Agree too.

```suggestion
Would you like to try updating your solution to use that?
```

Reply:
> (Of course, the answer may be "No, I wouldn't". That's never happened yet, though.)

On the JS track we get this _quite frequently_, which is why I stopped phrasing it like that.

Reply:
> How does using that improve the code?

This is a perfectly sensible and nice suggestion. However, students tend to take such questions a bit too seriously, and I don't like the idea of someone thinking too hard about this (remember, this is the very first program they've ever written in Go).

We've talked a lot about the string concatenation issue and how much information and nuance it's wise to load students with on this first exercise. I favour a light touch here; if they use `fmt.Sprintf`, great. If not, I'll let them know that it exists and it's a powerful way to format data into strings. If they're not particularly interested now, they'll learn about it soon enough.

I'd ask if they'd like to try updating their solution to use `fmt.Sprintf`, and leave it at that.

Reply:
For comparison, here's what Exalysis says if the student uses `+=` to concatenate strings. As usual, @tehsphinx strikes a very warm and genial note:

> I see you are using `+` to concatenate the string. That is perfectly legit! If you didn't know about the `fmt` package, check it out. It can do lots of things related to string formatting, including concatenating strings.
>
> (Using `+` is probably the right choice here. I just wanted you to know about `fmt`, because it's awesome, and you'll be using it a lot.)

Reply:
I don't think that was entirely me...

Reply:
> This is a perfectly sensible and nice suggestion. However, students tend to take such questions a bit too seriously, and I don't like the idea of someone thinking too hard about this

I feel the same way.

Reply:
I feel like being explicit that we expect an action is important as I too have had lots of people say "no" when I phrase things as "would you like to.." and have had essays rather than code back when I ask questions that don't explicitly tell people to submit code.

How about we make it more explicit that we want/expect someone to do something, but still polite (I'm english - it's what we do :))

```suggestion
Could you try updating your solution to use `fmt.Sprintf` please?
```